Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp

```diff
@@ -43,11 +43,36 @@ namespace Platform
         }
         return physicalDeviceList;
     }
+    
+    float GetRefreshRate()
+    {
+        CGDirectDisplayID display = CGMainDisplayID();
+        CGDisplayModeRef currentMode = CGDisplayCopyDisplayMode(display);
+        return CGDisplayModeGetRefreshRate(currentMode);
+    }
 
-    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval)
+    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval, float refreshRate)
     {
-        AZ_UNUSED(syncInterval);
-        [mtlCommandBuffer presentDrawable:drawable];
+        bool hasPresentAfterMinimumDurationAPI = false;
+        bool framePresented = false;
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
+        
+#if defined(__MAC_11_0)
+        if(@available(iOS 14.0, macOS 11.0, *))
+        {
+            hasPresentAfterMinimumDurationAPI = [drawable respondsToSelector:@selector(presentAfterMinimumDuration:)];
+            if (hasPresentAfterMinimumDurationAPI && presentAfterMinimumDuration > 0.0f)
+            {
+                [mtlCommandBuffer presentDrawable:drawable afterMinimumDuration:presentAfterMinimumDuration];
```

Comment:
This looks wrong. The selector should match the name of the obj-c function you call.

Reply:
Reading the docs it seems [MTLCommandBuffer presentDrawable: afterMinimumDuration:] is a utility function that forwards to [MTLDrawable presentAfterMinimumDuration] at the correct time.

Please add a comment explaining why you check for one function but call the other.